cmd/arc_summary/arc_summary.py

```diff
@@ -705,8 +705,8 @@ def _l2arc_summary(Kstat):
         )
         sys.stdout.write("\n")
 
-        if arc["l2_arc_evicts"]['lock_retries'] + \
-                arc["l2_arc_evicts"]["reading"] > 0:
+        if arc["l2_arc_evicts"]['lock_retries'] != '0' or \
+           arc["l2_arc_evicts"]["reading"] != '0':
```

Comment:
That would would be correct, yes. Right now we've got _l2arc_summary() doing the actual writing of the human readable l2arc summary output, based on data it gets from get_l2arc_summary(). However, get_l2arc_summary() does a bunch of the formatting of the data as well right now, including calling fHits() and fPerc() to add units to numbers etc.

If it was up to me we'd flip that such that get_l2arc_summary() gathers the data into a dictionary, including maybe computing some percentages, and returning that. Then _l2arc_summary() would do the writing of the output, including adding units to numbers etc.

If you want a stab at a change like that I'd be happy to propose one.